### ISSUE TITLE
The failure callback will be executed if the success callback of `denops#request_async()` throws an exception.

### ISSUE THREAD

**Issue by tennashi** [vim-denops/denops.vim#114]
* `denops/sample/main.ts`
```ts
import { Denops } from "https://deno.land/x/denops_std/mod.ts";

export function main(denops: Denops): Promise<void> {
  denops.dispatcher = {
    test(): Promise<string> {
      return Promise.resolve("test done")
    }
  }
  return Promise.resolve();
}
```

* `plugin/sample.vim`
```vim
function! Hoge()
  call denops#request_async("sample", "test", [], function("s:on_success"), function("s:on_failure"))
endfunction

function! s:on_success(...)
  echom a:000
  throw "hoge"
endfunction

function! s:on_failure(...)
  echom a:000
endfunction
```

When I wrote the above and executed the `Hoge()` function, I got the following output

```
:call Hoge()
s:on_success
['test done']
s:on_failure
[{}]
```

The empty output after on_failure is due to issue #112.

**Comment by lambdalisue** [comment]
Could you check if #113 fix the issue as well? I rewrote that PR and fix both #112 and #114 @tennashi

**Comment by tennashi** [comment]
I tried it.

The `s:on_failure` callback is no longer called, but the exception is not what I expected.

I got:
```
[denops] Error: Failed to call 'denops#callback#call(276e6b510b822d3ddfd0f04b8cdb9b7fc7ca994fc25980dd64bd2764cd83d6df, test done)':
 vim exception
[denops]     at Vim.call (file:///home/tennashi/src/github.com/vim-denops/denops.vim/denops/@denops-private/host/vim.ts:76:13)
[denops]     at async Service.call (file:///home/tennashi/src/github.com/vim-denops/denops.vim/denops/@denops-private/service.ts:97
:12)
[denops]     at async file:///home/tennashi/src/github.com/vim-denops/denops.vim/denops/@denops-private/host/invoker.ts:45:11
```

I want:
```
E605 Exception not caught: hoge
```
